tensorzero-core/src/variant/chat_completion.rs

```diff
@@ -50,6 +50,83 @@ pub struct ChatTemplates {
     pub assistant: Option<TemplateWithSchema>,
 }
 
+impl ChatTemplates {
```

Comment:
can you add a little top level comment about where this is called / why it exists